src/query/execute_statement.c

```diff
@@ -2846,6 +2860,28 @@ do_alter_serial (PARSER_CONTEXT * parser, PT_NODE * statement)
 	  goto end;
 	}
       pr_clear_value (&value);
+      opt_serial_option_list_flag = true;
+    }
+
+  /* owner to */
+  serial_owner = statement->info.serial.owner_name;
+  if (serial_owner != NULL)
+    {
```

Comment:
Given that the existence of the `owner_name` node is guaranteed during the syntax check, this condition doesn't have to be checked. It would be more suitable to add `assert(serial_owner != NULL)` instead of the if-statement.

Reply:
Thank you for the code review.

As you mentioned, the existence of the owner_name node is guaranteed by performing a syntax check in advance.

The reason for adding the if-statement is as follows.

In the ALTER SERIAL syntax, both options below are performing the do_alter_serial() function.
Option 1) opt_serial_option_list(INCREMENT BY, START WITH, MINVALUE, MAXVALUE, CYCLE, CACHE)
Option 2) OWNER TO

Therefore, in the do_alter_serial() function, an if statement was added to check whether the OWNER TO option is used.

Reply:
It seems that the behavior can be controlled by `opt_serial_option_list_flag` that you introduced.

Reply:
I revised it by referring to the offline review. Please confirm.

1) Added semantic check to PT_ALTER_SERIAL
  - A semantic check was performed along with the existing syntax check, but a semantic check was added to operate according to meaning in each session.

2) We have modified the duplicate inspection part.
  - Fixed a part in the execution session that checks for the same owner_name.